src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2567,6 +2594,15 @@ public ConstructorEdit(INamedTypeSymbol oldType)
                                     var containingSymbolKey = SymbolKey.Create(newContainingType, cancellationToken);
                                     oldContainingType = containingSymbolKey.Resolve(oldCompilation, ignoreAssemblyKey: true, cancellationToken).Symbol as INamedTypeSymbol;
 
+                                    if (oldContainingType != null && !capabilities.HasCapability(ManagedEditAndContinueCapability.AddDefinitionToExistingType))
+                                    {
+                                        diagnostics.Add(new RudeEditDiagnostic(
+                                            RudeEditKind.Insert,
```

Comment:
>RudeEditKind.Insert [](http://example.com/codeflow?start=44&length=19)

I wonder if we should use a different RudeEditKind so we are able to separate telemetry data.

Reply:
I was wondering that too. Well, that's not true, I was wondering if we should have a different message, but then I thought that runtime capabilities is an implementation detail the user can't do much about, but I guess a different kind for the same message doesn't hurt.

Reply:
The message can be the same but it might actually be useful for the user to see this is a runtime limitation, so that they are not wondering why it works when doing the same thing in on .NET Core, e.g.